client/src/parsec/terms_of_service.ts

```diff
@@ -0,0 +1,40 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+import { needsMocks } from '@/parsec/environment';
+import { getParsecHandle } from '@/parsec/routing';
+import { ClientAcceptTosError, ClientGetTosError, Result, Tos } from '@/parsec/types';
+import { libparsec } from '@/plugins/libparsec';
+import { DateTime } from 'luxon';
+
+export async function acceptTOS(updatedOn: DateTime): Promise<Result<null, ClientAcceptTosError>> {
+  const handle = getParsecHandle();
+
+  if (handle !== null && !needsMocks()) {
+    return await libparsec.clientAcceptTos(handle, updatedOn.toSeconds() as any as DateTime);
+  } else {
+    return { ok: true, value: null };
+  }
+}
+
+export async function getTOS(): Promise<Result<Tos, ClientGetTosError>> {
+  const handle = getParsecHandle();
+
+  if (handle !== null && !needsMocks()) {
```

Comment:
I'm not familiar with the client code: No handle means there is an error in the application? I suppose it should not happen, but could there be an error that makes `handle == null`? In that case, what is presented to the user?

Why not handling it like:
```ts
if (handle !== null) {
    // error case
}

if (!needsMocks()){
    // normal case
} else {
    // mock case
}
```

Reply:
No handle shouldn't be possible unless the user messes with the URL, which is not possible with Electron (only in web mode).
I'll have to do a big pass on everything related to how we mock data and how to handle these kind of errors once we got the web bindings.